TopMovies/API/GraphQL/QueryProducer.swift

```diff
@@ -0,0 +1,132 @@
+//
+//  QueryProducer.swift
+//  TopMovies
+//
+//  Created by Macbook Pro  on 01.04.2021.
+//
+
+struct QueryProducer {
+  private enum CategoryQueryName: String, CaseIterable {
+    case nowPlaying, popular, topRated, upcoming
+    
+    init(categoryRequest: MovieCategoryRequest) {
+      switch categoryRequest {
+      case .nowPlaying: self = .nowPlaying
+      case .popular: self = .popular
+      case .topRated: self = .topRated
+      case .upcoming: self = .upcoming
+      }
+    }
+  }
+  
+  private let moviePreviewQuery =
+    """
+    id
+    title
+    overview
+    poster(size: W780)
+    backdrop(size: W780)
+    isAdult
+    releaseDate
+    rating
+    """
+  
+  func allCategoriesQuery() -> String {
+    var queryString =
+      """
+      {
+        movies {
+      """
+        CategoryQueryName.allCases
```

Comment:
I'd suggest moving all cases array in `allCategoriesQuery` param and provide `ategoryQueryName.allCases` in the method itself. You would need only 1 method, that will work for 1 category and for multiple categories.